enhancements/console/catalog-categories.md

```diff
@@ -0,0 +1,153 @@
+
+## Proposal
+
+### Extending the Console CRD
+
+Extend the existing `operator.openshift.io/v1` / `Console` CRD. It provides already a `spec.customization` area where we can add a list of available categories.
+
+Each category could have a list of subcategories which enables the Developer Catalog to build a 2 level tree based on this data:
+
+```
+├── Languages
+│   ├── Java
+│   ├── JavaScript
+│   └── ...
+├── Databases
+├── Middleware
+│   ├── Integration
+│   ├── Process Automation
+│   └── ...
+└── ...
+```
+
+Each Category and Subcategories matches following schema:
+
+* `id: string`, identifier used in the URL to enable deeplinking in console
+* `label: string`, category display label
+* `disabled: boolean`, flag to hide this category from the sidebar so that the admin is not required to remove a complete section to hide an unused category
+* `tags: string[]`, the list of item tags this category will match
+
+Categories on the first level contains also a list of subcategories:
+
+* `subcategories:` a list of Categories
+
+If a category has subcategories the dev console will show all tags from a selected category and all subcategories.
+
+So the final yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  annotations:
+    release.openshift.io/create-only: 'true'
+  name: cluster
+  ...
+spec:
+  customization:
+    brand: online
+    devcatalog:
+      categories:
+      - id: languages
+        label: Languages
+        disabled: false
+        tags: [go]
+        subcategories:
+        - id: java
+          label: Java
+          tags: [java]
+        - id: javascript
+          tags: [javascript, nodejs, js]
+  ...
+```
+
+### User Stories
+
+#### Story 1
+
+As a admin, I want to add, update or remove some of the catalog categories.
+
+#### Story 2
+
+As a user, I want to use the configured catalog categories in the Developer Catalog.
+
+### Risks and Mitigations
+
+## Open Questions
+
+- [ ] Is the `Console` CRD the best option, or should we create a new `ConsoleCategory` CRD instead? Pro and con: Operators can provide CRs
+- [ ] How can we provide a default list. Can we automatically create it when there is currently no devcatalog categories defined? So the admin can restore the default value by removing the current configuration?
+
+### Test Plan
+
+Testing will be carried out with the usual Console unit and e2e test suites.
+
+### Graduation Criteria
+
+None
+
+### Upgrade / Downgrade Strategy
+
+When updating to 4.7 we need to ensure that the default types was automatically added to the existing Console CR.
+
+Downgrading from 4.7 to an earlier version would ignore this additional field and would show the old static categories to the user.
+
+### Version Skew Strategy
+
+> TODO: Christoph
+
+## Implementation History
+
+None
+
+## Drawbacks
+
+- A bigger `Console` CR must fetches by the console before the Developer Catalog is opened.
```

Comment:
We wouldn't be fetching the CR in the frontend code. The operator will watch the config, and then wire through the values to the console backend through its config map in the openshift-console namespace.

Reply:
Thanks :+1: Changed the document to make it clear that the frontend reads the data from the ConfigMap.